static/src/javascripts/projects/commercial/modules/header-bidding/a9/a9.ts

```diff
@@ -1,14 +1,19 @@
 import config from '../../../../../lib/config';
-import { getHeaderBiddingAdSlots } from '../slot-config';
+import type { Advert } from '../../dfp/Advert';
 import { dfpEnv } from '../../dfp/dfp-env';
+import { getHeaderBiddingAdSlots } from '../slot-config';
 
 /*
  * Amazon's header bidding javascript library
  * https://ams.amazon.com/webpublisher/uam/docs/web-integration-documentation/integration-guide/javascript-guide/display.html
  */
 
-class A9AdUnit {
-	constructor(advert, slot) {
+class A9AdUnit implements A9AdUnitInterface {
+	slotID?: string;
+	slotName?: string;
+	sizes?: HeaderBiddingSize[];
```

Comment:
I was looking for a way to have A9AdUnit inherit these properties but couldn't find one.
Any suggestions?

Reply:
I think this is the way that class implementation works. You have to explicitly declare these properties. You don’t have to make them conditional if you are setting them in the constructor, though!

Reply:
Good call